### PR TITLE
DOC Improve contributing instructions

### PR DESCRIPTION
**Pull request by rth**
More detailed instructions on how to contribute.

Also including @ryanking13  suggestions from the previous discussion and https://github.com/pyodide/pyodide/issues/2440

### CODE REVIEW

File: docs/development/contributing.md [hunk 1]
```diff
@@ -8,17 +8,56 @@ for diving into it.
 
 ## Development Workflow
 
-See {ref}`building_from_sources` and {ref}`testing` documentation.
+To contribute code, see the following steps,
 
-For code-style the use of [pre-commit](https://pre-commit.com/) is also recommended,
+1. Fork the Pyodide repository [https://github.com/pyodide/pyodide](https://github.com/pyodide/pyodide) on Github.
+2. If you are on Linux, you can skip this step. On Windows and MacOS you have a
+   choice between, manually installing Docker,
```
Comment (hoodmane) [inline docs/development/contributing.md:15]:
```suggestion
   choice. The first option is to manually install Docker:
```

File: docs/development/contributing.md [hunk 2]
```diff
@@ -8,17 +8,56 @@ for diving into it.
 
 ## Development Workflow
 
-See {ref}`building_from_sources` and {ref}`testing` documentation.
+To contribute code, see the following steps,
 
-For code-style the use of [pre-commit](https://pre-commit.com/) is also recommended,
+1. Fork the Pyodide repository [https://github.com/pyodide/pyodide](https://github.com/pyodide/pyodide) on Github.
+2. If you are on Linux, you can skip this step. On Windows and MacOS you have a
```
Comment (hoodmane) [inline docs/development/contributing.md:14]: It's still recommended to use docker in linux though?

File: docs/development/contributing.md [hunk 3]
```diff
@@ -8,17 +8,56 @@ for diving into it.
...
-pre-commit install
-```
+   - on MacOS follow [these instructions](https://docs.docker.com/desktop/mac/install/)
+   - on Windows, [install WSL
+     2](https://docs.microsoft.com/en-us/windows/wsl/install), then Docker.
+     Note that Windows filesystem access from WSL2 is very slow and should
+     be avoided when building Pyodide.
+
+   The other alternative is to use a service that provides a Linux
+   development environment, such as,
```
Comment (hoodmane) [inline docs/development/contributing.md:24]: There should either be a colon or no punctuation here and in other places before display code.
```suggestion
   development environment, such as
```

File: docs/development/contributing.md [hunk 4]
```diff
@@ -8,17 +8,56 @@ for diving into it.
...
+
+   - [Github Codespaces](https://github.com/features/codespaces)
+   - [gitpod.io](https://gitpod.io)
+   - or a remote Linux VM with SSH connection.
+
+3. Clone your fork of Pyodide,
```
Comment (hoodmane) [inline docs/development/contributing.md:30]:
```suggestion
3. Clone your fork of Pyodide
```

File: docs/development/contributing.md [hunk 5]
```diff
@@ -8,17 +8,56 @@ for diving into it.
...
-pip install pre-commit
-pre-commit install
-```
+   - on MacOS follow [these instructions](https://docs.docker.com/desktop/mac/install/)
+   - on Windows, [install WSL
+     2](https://docs.microsoft.com/en-us/windows/wsl/install), then Docker.
+     Note that Windows filesystem access from WSL2 is very slow and should
+     be avoided when building Pyodide.
+
+   The other alternative is to use a service that provides a Linux
```
Comment (hoodmane) [inline docs/development/contributing.md:23]:
```suggestion
   The second option is to use a service that provides a Linux
```

**Review by hoodmane — APPROVED** [review]
Thanks @rth!

File: docs/development/contributing.md (hunk 2 above)
Comment (rth) [reply docs/development/contributing.md:14]: Yeah, but we already say that in the section about building from sources. So I think it's fine.

**Pull request merged**